ReactSkia/components/RSkComponentTextInput.cpp

```diff
@@ -236,32 +287,118 @@ void RSkComponentTextInput::onHandleKey(rnsKey eventKeyType, bool *stopPropagati
     textInputEventEmitter->onChange(textInputMetrics);
     textInputEventEmitter->onContentSizeChange(textInputMetrics);
     textInputEventEmitter->onSelectionChange(textInputMetrics);
-  }//else if (isInEditingMode_)
+}
+
+void RSkComponentTextInput::KeyEventProcessingThread(){
+  std::string textString = {};
+  struct cursor lCursor;
+  RNS_LOG_DEBUG("Creating the thread worker thread");
+  KeyPressMetrics keyPressMetrics;
+  TextInputMetrics textInputMetrics;
+  auto component = getComponentData();
+  auto textInputEventEmitter = std::static_pointer_cast<TextInputEventEmitter const>(component.eventEmitter);
+  while(isThreadAlive){
+    if(!inputQueue.empty()){
+      globalVarProtectorMutex.lock();
+      lCursor.locationFromEnd = cursor_.locationFromEnd;
+      textString = displayString_;
+      auto eventKeyType = inputQueue.front();
+      inputQueue.pop();
+      globalVarProtectorMutex.unlock();
+      lCursor.end = textString.length();
+      keyPressMetrics.text = RNSKeyMap[eventKeyType];
```

Comment:
we can call eventKeyProcessing() here right , since its duplicate code of this API